hs_core/views/utils.py

```diff
@@ -1027,74 +1027,71 @@ def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original,
+        # walk through each unzipped file, delete aggregations if they exist
+        for file in res_files:
+            destination_file = _get_destination_filename(file.name, unzipped_foldername)
+            if (istorage.exists(destination_file)):
+                if not overwrite:
+                    override_tgt_paths.append(destination_file.split('data/contents/')[1])
+                elif resource.resource_type == "CompositeResource":
+                    aggregation_object = resource.get_file_aggregation_object(
+                        destination_file)
+                    if aggregation_object:
+                        aggregation_object.logical_delete(user)
                     else:
+                        logger.error("No aggregation object found for " + destination_file)
                         istorage.delete(destination_file)
-            # now move each file to the destination
-            for file in res_files:
-                destination_file = _get_destination_filename(file.name, unzipped_foldername)
-                istorage.moveFile(file.name, destination_file)
-            # and now link them to the resource
-            added_resource_files = []
-            for file in res_files:
-                destination_file = _get_destination_filename(file.name, unzipped_foldername)
-                destination_file = destination_file.replace(res_id + "/", "")
-                destination_file = resource.get_irods_path(destination_file)
-                res_file = link_irods_file_to_django(resource, destination_file)
-                added_resource_files.append(res_file)
-
-            if resource.resource_type == "CompositeResource":
-                # make the newly added files part of an aggregation if needed
-                for res_file in added_resource_files:
-                    resource.add_file_to_aggregation(res_file)
-
-            if auto_aggregate:
-                check_aggregations(resource, added_resource_files)
-            if ingest_metadata:
-                # delete original zip to prevent from being pulled into an aggregation
-                zip_with_rel_path = zip_with_rel_path.split("contents/", 1)[1]
-                delete_resource_file(res_id, zip_with_rel_path, user)
-
-                from hs_file_types.utils import ingest_metadata_files
-                ingest_metadata_files(resource, meta_files, map_files)
-            istorage.delete(unzip_path)
-        else:
-            unzip_path = istorage.unzip(zip_with_full_path)
-            res_files = link_irods_folder_to_django(resource, istorage, unzip_path, auto_aggregate)
-            if resource.resource_type == 'CompositeResource':
-                # make the newly added files part of an aggregation if needed
-                for res_file in res_files:
-                    resource.add_file_to_aggregation(res_file)
+                else:
+                    istorage.delete(destination_file)
+        if not overwrite and override_tgt_paths:
```

Comment:
Should this check happen before all the aggregation manipulations that come before?

Reply:
ok, I see that now on line 1052. The overwrite check gets lost in all of it. We can reduce some of the nesting if `override_tgt_paths` is always created.  Something like this:
```
for file in res_files:
  destination_file = _get_destination_filename(file.name, unzipped_foldername)
  if (istorage.exists(destination_file)):
    if not overwrite:
      override_tgt_paths.append(destination_file.split('data/contents/')[1])
if not overwrite and override_tgt_paths:
  message = 'move would overwrite {}'.format(', '.join(override_tgt_paths))
  raise FileOverrideException(message)
for override_tgt_path in override_tgt_paths:
  aggregation_object = resource.get_file_aggregation_object(destination_file)
  if aggregation_object:
    aggregation_object.logical_delete(user)
  else:
    istorage.delete(destination_file)
```